src/ethernet/ip_device.cpp

```diff
@@ -0,0 +1,310 @@
+
+    ((RsRTSPClient *)remote_sensors[sensor_index]->rtsp_client)->initFunc(&rs_rtp_stream::get_memory_pool());
+    ((RsRTSPClient *)remote_sensors[sensor_index]->rtsp_client)->getStreams();
+}
+
+ip_device::~ip_device()
+{
+    is_device_alive = false;
+
+    for (int remote_sensor_index = 0; remote_sensor_index < NUM_OF_SENSORS; remote_sensor_index++)
+    {
+        stop_sensor_streams(remote_sensor_index);
+    }
+
+    if (sw_device_status_check.joinable())
+    {
+        sw_device_status_check.join();
+    }
+    std::cout << "destroy ip_device\n";
+}
+
+void ip_device::stop_sensor_streams(int sensor_index)
+{
+    for (long long int key : remote_sensors[sensor_index]->active_streams_keys)
+    {
+        std::cout << "\t@@@ stopping stream [uid:key] " << streams_collection[key].get()->m_rs_stream.uid << ":" << key << "]" << std::endl;
+        streams_collection[key].get()->is_enabled = false;
+        if (inject_frames_thread[key].joinable())
+            inject_frames_thread[key].join();
+    }
+    remote_sensors[sensor_index]->active_streams_keys.clear();
+}
+
+ip_device::ip_device(std::string ip_address, rs2::software_device sw_device)
+{
+    this->ip_address = ip_address;
+    this->is_device_alive = true;
+
+    //init device data
+    init_device_data(sw_device);
+}
+
+std::vector<rs2_video_stream> ip_device::query_streams(int sensor_id)
+{
+    std::cout << "Querry Sensors\n";
+    std::vector<rs2_video_stream> streams;
+
+    if (remote_sensors[sensor_id]->rtsp_client == NULL)
+        return streams;
+
+    //workaround
+    if (remote_sensors[sensor_id]->rtsp_client == nullptr)
+        recover_rtsp_client(sensor_id);
+
+    streams = remote_sensors[sensor_id]->rtsp_client->getStreams();
+    return streams;
+}
+std::vector<IpDeviceControlData> ip_device::get_controls(int sensor_id)
+{
+    std::cout << "GetControls\n";
+    std::vector<IpDeviceControlData> controls;
+    controls = remote_sensors[sensor_id]->rtsp_client->getControls();
+
+    return controls;
+}
+
+bool ip_device::init_device_data(rs2::software_device sw_device)
+{
+    std::string url, sensor_name = "";
+    for (int sensor_id = 0; sensor_id < NUM_OF_SENSORS; sensor_id++)
+    {
+
+        url = std::string("rtsp://" + ip_address + ":8554/" + sensors_str[sensor_id]);
+        sensor_name = sensors_str[sensor_id] + std::string(" (Remote)");
+
+        remote_sensors[sensor_id] = new ip_sensor();
+
+        remote_sensors[sensor_id]->rtsp_client = RsRTSPClient::getRtspClient(url.c_str(), "ip_device_device");
+        ((RsRTSPClient *)remote_sensors[sensor_id]->rtsp_client)->initFunc(&rs_rtp_stream::get_memory_pool());
+
+        rs2::software_sensor tmp_sensor = sw_device.add_sensor(sensor_name);
+
+        remote_sensors[sensor_id]->sw_sensor = std::make_shared<rs2::software_sensor>(tmp_sensor);
```

Comment:
Please change sw_sensor to rs2::software_sensor (instead of shared_ptr)

Reply:
Cannot do it since there is no default constructor for sw_sensor